api/src/routes/channels/#channel_id/index.ts

```diff
@@ -30,8 +30,8 @@ router.delete("/", route({ permission: "MANAGE_CHANNELS" }), async (req: Request
 	} else if (channel.type === ChannelType.GROUP_DM) {
 		await Channel.removeRecipientFromChannel(channel, req.user_id)
 	} else {
-		//TODO messages in this channel should be deleted before deleting the channel
 		await Promise.all([
+			Message.delete({ channel_id: channel_id }),
```

Comment:
you need to delete Message before the channel and therefore need to execute outside before Promise.all.
In fact you need to delete Invite, ReadState, VoiceState, Recipient, ... 
This will be done in https://github.com/fosscord/fosscord-server/pull/382

Reply:
Alright
Imma just go look at all references to a channel real quick lol

Reply:
redundant if #382 merges